Dockerfile

```diff
@@ -0,0 +1,39 @@
+# syntax=docker/dockerfile:1
+
+# Stage 1 - Build the toolchain
+FROM ubuntu:18.04
+ARG N64_INST=/n64_toolchain
+ENV N64_INST=${N64_INST}
+
+# install dependencies
+RUN apt-get update
+RUN apt-get install -yq wget bzip2 gcc g++ make file libmpfr-dev libmpc-dev zlib1g-dev texinfo git gcc-multilib
+
+# Build
+COPY ./tools/build-toolchain.sh /tmp/tools/build-toolchain.sh
+WORKDIR /tmp/tools
+RUN ./build-toolchain.sh
+
+# Strip executables
+RUN find ${N64_INST}/bin -type f | xargs strip
+RUN strip ${N64_INST}/libexec/gcc/mips64-elf/10.2.0/plugin/gengtype
+RUN strip ${N64_INST}/libexec/gcc/mips64-elf/10.2.0/liblto_plugin.so.0.0.0
+RUN strip ${N64_INST}/libexec/gcc/mips64-elf/10.2.0/liblto_plugin.so.0
+RUN strip ${N64_INST}/libexec/gcc/mips64-elf/10.2.0/lto-wrapper
+RUN strip ${N64_INST}/libexec/gcc/mips64-elf/10.2.0/collect2
+RUN strip ${N64_INST}/libexec/gcc/mips64-elf/10.2.0/cc1plus
+RUN strip ${N64_INST}/libexec/gcc/mips64-elf/10.2.0/cc1
+RUN strip ${N64_INST}/libexec/gcc/mips64-elf/10.2.0/install-tools/fixincl
+RUN strip ${N64_INST}/libexec/gcc/mips64-elf/10.2.0/lto1
+RUN rm -rf ${N64_INST}/share/locale/*
+
+# Stage 2 - Prepare minimal image
+FROM ubuntu:18.04
+ARG N64_INST=/n64_toolchain
+ENV N64_INST=${N64_INST}
+
+COPY --from=0 ${N64_INST} ${N64_INST}
+RUN apt-get update && \
+    apt-get install -yq gcc g++ make libpng-dev git  && \
+    apt-get clean && \
+    apt autoremove -yq
```

Comment:
I think if you switch these two statements (first RUN then COPY), it should improve caching, because the COPY will possibly change everytime libdragon changes, but the way it is right now also RUN will be re-run every time.

Reply:
You are right but I don't think it will matter too much b/c if we have already built the toolchain it already takes a lot of time. This might be negligable :) Also arguably it does also update packages to the latest if distro lists are updated?